lib/src/components/NxAccordion/__tests__/NxAccordion.test.tsx

```diff
@@ -5,277 +5,221 @@
+  const renderEl = rtlRenderElement(NxAccordion, {});
 
   it('renders a <details> element with the provided props', function() {
-    const component = getShallowComponent({ id: 'foo', open: true, title: 'test title' });
-
-    expect(component).toMatchSelector('details#foo');
-    expect(component).toHaveProp('open', true);
-    expect(component).toHaveProp('title', 'test title');
+    expect(quickRender({ id: 'foo' }).getByRole('group')).toHaveAttribute('id', 'foo');
+    expect(renderEl({ open: true })).toHaveAttribute('open');
+    expect(renderEl({ title: 'test title' })).toHaveAttribute('title', 'test title');
   });
 
-  it('sets the nx-accordion class along with any provided className', function() {
-    expect(getShallowComponent()).toHaveClassName('nx-accordion');
-    expect(getShallowComponent({ className: 'foo' })).toHaveClassName('nx-accordion');
-    expect(getShallowComponent({ className: 'foo' })).toHaveClassName('foo');
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
   it('sets an id if none is specified', function() {
-    expect(getShallowComponent().prop('id')).toBeTruthy();
+    expect(quickRender().getByRole('group')).toHaveAttribute('id');
   });
 
   it('sets aria-expanded from the open prop', function() {
-    expect(getShallowComponent()).toHaveProp('aria-expanded', false);
-    expect(getShallowComponent({ open: false })).toHaveProp('aria-expanded', false);
-    expect(getShallowComponent({ open: true })).toHaveProp('aria-expanded', true);
+    expect(renderEl()).toHaveAttribute('aria-expanded', 'false');
+    expect(renderEl({ open: false })).toHaveAttribute('aria-expanded', 'false');
+    expect(renderEl({ open: true })).toHaveAttribute('aria-expanded', 'true');
   });
 
   it('renders non-header children in an nx-accordion__content wrapper', function() {
-    const component =
-        mount(
-          <NxAccordion>
-            <NxAccordion.Header>
-              <span>Foo</span>
-            </NxAccordion.Header>
-            <span>Bar</span>
-          </NxAccordion>
-        ),
-        childrenWrapper = component.find('.nx-accordion__content');
-
-    expect(childrenWrapper).toExist();
-    expect(childrenWrapper).toContainReact(<span>Bar</span>);
-    expect(childrenWrapper).not.toContainReact(<span>Foo</span>);
+    const { rerender, container } = quickRender();
+
+    rerender(
+      <NxAccordion>
+        <NxAccordion.Header>
+          <span className='foo'>Foo</span>
+        </NxAccordion.Header>
+        <span className='bar'>Bar</span>
+      </NxAccordion>
+    );
+
+    expect(container.querySelector('.nx-accordion__header .foo')).toBeInTheDocument();
+    expect(container.querySelector('.nx-accordion__content .bar')).toBeInTheDocument();
```

Comment:
Rather than testing that it's in `.nx-accordion__content`, I'd test that it's in the accordion in general, and that it's not in the `<summary>` element

Reply:
> Rather than testing that it's in `.nx-accordion__content`, I'd test that it's in the accordion in general, and that it's not in the `<summary>` element

Like this? ad3b1d55d33707c24cabbaad0ac7cddeae469eab

Reply:
> Like this? https://github.com/sonatype/sonatype-react-shared-components/commit/ad3b1d55d33707c24cabbaad0ac7cddeae469eab

Close but not quite.  `.nx-accordion` is also an implementation detail.  I'd just check that `.bar` is present in the rendered output at all (in addition to checking that `summary .bar` isn't)